internal/csi-addons/rbd/replication.go

```diff
@@ -115,6 +121,30 @@ func getForceOption(ctx context.Context, parameters map[string]string) (bool, er
 	return force, nil
 }
 
+// getFlattenMode gets flatten mode from the input GRPC request parameters.
+// flattenMode is the key to check the mode in the parameters.
+func getFlattenMode(ctx context.Context, parameters map[string]string) corerbd.FlattenMode {
+	val, ok := parameters[flattenModeKey]
+	if !ok {
+		log.DebugLog(ctx, "%s is not set in parameters, setting to default (%v)",
+			flattenModeKey, corerbd.FlattenModeNever)
+
+		return corerbd.FlattenModeNever
+	}
+
+	switch corerbd.FlattenMode(val) {
```

Comment:
```suggestion
mode := corerbd.FlattenMode(val);
switch  mode {
	case corerbd.FlattenModeForce, corerbd.FlattenModeNever:
		return mode,nil
	default:
		log.DebugLog(ctx, "flattenMode %v not supported",mode)
		return mode,err
	}
```
lets return error in case of its a user mistake instead of choosing FlattenModeNever